src/Elliptic/Systems/Poisson/Equations.cpp

```diff
@@ -0,0 +1,182 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "Elliptic/Systems/Poisson/Equations.hpp"
+
+#include "DataStructures/DataVector.hpp"
+#include "DataStructures/Index.hpp"
+#include "DataStructures/Matrix.hpp"
+#include "DataStructures/Tensor/Tensor.hpp"
+#include "DataStructures/Variables.hpp"
+#include "Elliptic/Systems/Poisson/MixedSystem.hpp"
+#include "Elliptic/Systems/Poisson/Tags.hpp"
+#include "NumericalAlgorithms/LinearOperators/Divergence.hpp"
+#include "NumericalAlgorithms/LinearOperators/PartialDerivatives.hpp"
+#include "Utilities/GenerateInstantiations.hpp"
+#include "Utilities/TMPL.hpp"
+
+namespace Poisson {
+
+template <size_t Dim>
+void ComputeMixedVolume<Dim>::apply(
+    const gsl::not_null<Scalar<DataVector>*> volume_for_field_source,
+    const gsl::not_null<tnsr::I<DataVector, Dim, Frame::Inertial>*>
+        volume_for_auxiliary_field_source,
+    const tnsr::i<DataVector, Dim, Frame::Inertial>& grad_field,
+    const tnsr::I<DataVector, Dim, Frame::Inertial>& auxiliary_field,
+    const Mesh<Dim>& mesh,
+    const InverseJacobian<DataVector, Dim, Frame::Logical, Frame::Inertial>&
+        inverse_jacobian) noexcept {
+  auto div_vars = make_with_value<Variables<tmpl::list<AuxiliaryField<Dim>>>>(
+      auxiliary_field, 0.);
+  get<AuxiliaryField<Dim>>(div_vars) = auxiliary_field;
+  *volume_for_field_source = get<Tags::div<AuxiliaryField<Dim>>>(
```

Comment:
we should probably let `ComputeDiv` act on a subset of the Variables just like the partial derivatives; that doesn't need to be done for this PR

Reply:
#966